src/exercises/day-1/luhn.rs

```diff
@@ -16,31 +16,32 @@
 // ANCHOR: luhn
 pub fn luhn(cc_number: &str) -> bool {
     // ANCHOR_END: luhn
+    let cc_number = cc_number.replace(" ", "");
     let mut sum = 0;
-    let mut double = false;
-    let mut digit_seen = 0;
+    let mut even = false;
 
-    for c in cc_number.chars().filter(|&f| f != ' ').rev() {
-        if let Some(digit) = c.to_digit(10) {
-            if double {
-                let double_digit = digit * 2;
-                sum += if double_digit > 9 {
-                    double_digit - 9
-                } else {
-                    double_digit
-                };
-            } else {
-                sum += digit;
+    if cc_number.len() < 2 {
+        return false;
+    }
+
+    for c in cc_number.chars().rev() {
+        let mut num;
+
+        match c.to_digit(10) {
+            Some(n) => num = n,
+            None => return false,
+        }
```

Comment:
Please use the `if let Some(num) = c.to_digit(10) else return false` pattern here.